cmd/dsl/util.go

```diff
@@ -50,3 +51,33 @@ func readTest(filename string) []string {
 	}
 	return strings.Split(strings.TrimSpace(string(testfile)), "\n")
 }
+
+// matches (type post) (value.content hello) (channel ssb-help)
+var postPattern = regexp.MustCompile(`\((\S+)\s(.*?)\)`)
+
+/* TODO:
+* implement number parsing (strconv attempt), boolean parsing (if true|false => bool it up)
+ */
+func parsePostLine(line string) map[string]interface{} {
```

Comment:
strongly suggest to add tests for this. Will make refactoring and adding tests much easier.

might want to move this out of the main package. `internal/postparser` or something might be a good place.

https://git.sr.ht/~cryptix/go-ssb-refs/tree/master/item/refs_test.go#L14 could be an inspiration of how to set these up in a _table driven way_ (same structure, lots of cases).

In essenace you have a `testCase` type and then iterate over a slice of them. Here this could be:

```go
type testCase struct{
  Input string
  Expected map[string]interface{}
  ExpError err
}
```

with the stetcher assert you should be able to just assert for equal without any manual nested assertion hassle.